modules/management-console-agent/src/main/java/org/apache/ignite/internal/agent/action/controller/QueryActionsController.java

```diff
@@ -89,6 +89,19 @@ public QueryActionsController(GridKernalContext ctx) {
         return new IgniteFinishedFutureImpl<>();
     }
 
+    /**
+     * @param globalQryId Global query id.
+     * @return Kill query result.
+     */
+    public IgniteInternalFuture<?> kill(String globalQryId) {
+        return ctx.closure().runLocalSafe(() -> {
+            SqlFieldsQuery qryFields = new SqlFieldsQuery("KILL QUERY '" + globalQryId + "';")
+                .setLazy(false);
+
+            qryProc.querySqlFields(qryFields, true).close();
+        }, MANAGEMENT_POOL);
+    }
```

Comment:
May be it make sense to introduce base class for ActionsControllers?
And declare protected method "safeExecute(lambda)" that will accept function to execute?

Reply:
@nva What do you think about it? If yes then i create a separate ticket for this and change in all places  where it need.

Reply:
It will reduce number of boilerplate code. IMHO.

Reply:
Let's do this in separated ticket.

Reply:
Ticket: GG-28865